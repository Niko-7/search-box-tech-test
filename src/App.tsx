import React from "react";

export const App = () => {
  return <h1>Initial test</h1>;
};
