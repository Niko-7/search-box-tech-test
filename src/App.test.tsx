/*eslint-disable */
import { cleanup, render, screen } from "@testing-library/react";
import { App } from "./App";
import renderer from "react-test-renderer";
const { axe, toHaveNoViolations } = require("jest-axe");
expect.extend(toHaveNoViolations);

afterEach(() => {
  cleanup();
});

describe("App component", () => {
  it("should render App component", () => {
    render(<App />);
    expect(<App />).toBeInTheDocument;
  });
});

describe("accesibility tests", () => {
  it("should have no accesibility errors on app component", async () => {
    const { container } = render(<App />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Snapshot tests", () => {
  it("should match the snapshot of the App component", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
