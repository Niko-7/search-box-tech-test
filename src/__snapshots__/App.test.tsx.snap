// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests should match the snapshot of the App component 1`] = `
<div>
  <div
    className="main"
  >
    <div
      className="container"
    >
      <div
        className="searchbox"
      >
        <div
          className="searchbox-main-container"
        >
          <div
            className="searchwidget__input-container"
          >
            <span
              aria-hidden="true"
              className="icon"
              data-testid="icon-container"
              role="presentation"
            >
              <svg
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="M17.464 6.56a8.313 8.313 0 1 1-15.302 6.504A8.313 8.313 0 0 1 17.464 6.56zm1.38-.586C16.724.986 10.963-1.339 5.974.781.988 2.9-1.337 8.662.783 13.65c2.12 4.987 7.881 7.312 12.87 5.192 4.987-2.12 7.312-7.881 5.192-12.87zM15.691 16.75l7.029 7.03a.75.75 0 0 0 1.06-1.06l-7.029-7.03a.75.75 0 0 0-1.06 1.06z"
                />
              </svg>
            </span>
            <input
              aria-label="Pick-up Location"
              autoComplete="off"
              className="search-input"
              onChange={[Function]}
              placeholder="Pick-up Location"
              type="text"
            />
            <noscript />
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    aria-label="Information Footer"
  >
    <div
      className="information-footer"
    >
      <a
        data-alt="Linked In Page"
        href="https://www.linkedin.com/in/nfloros/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <p>
          Nikolaos Floros
        </p>
      </a>
      <p
        className="user-information"
      >
        Front-End Technical Test
      </p>
      <br />
      <a
        data-alt="Github Repo Source Code Page"
        href="https://github.com/Niko-7/search-box-tech-test"
        rel="noopener noreferrer"
        target="_blank"
      >
        <p>
          View Source Code
        </p>
      </a>
    </div>
  </footer>
</div>
`;
